compiler/qsc_parse/src/item/tests.rs

```diff
@@ -1557,3 +1557,139 @@ fn disallow_qubit_scoped_block() {
             ]"#]],
     );
 }
+
+#[test]
+fn reject_nested_namespace_with_items() {
+    check_vec(
+        parse_namespaces,
+        "namespace Outer {
+            namespace Inner {
+                function NestedFunction() : Unit {}
+                newtype NestedType = Int;
+            }
+        }",
+        &expect![[r#"
+            Namespace _id_ [0-99] (Ident _id_ [10-15] "Outer"):
+
+            [
+                Error(
+                    Token(
+                        Close(
+                            Brace,
+                        ),
+                        Keyword(
+                            Namespace,
+                        ),
+                        Span {
+                            lo: 30,
+                            hi: 39,
+                        },
+                    ),
+                ),
+                Error(
+                    Token(
+                        Eof,
+                        Keyword(
+                            Newtype,
+                        ),
+                        Span {
+                            lo: 116,
+                            hi: 123,
+                        },
+                    ),
+                ),
+            ]"#]],
+    );
+}
+
+#[test]
+fn reject_namespace_with_multiple_nested_levels() {
+    check_vec(
+        parse_namespaces,
+        "namespace LevelOne {
+            namespace LevelTwo {
+                namespace LevelThree {
+                    function DeepFunction() : Unit {}
+                }
+            }
+        }",
+        &expect![[r#"
+            Namespace _id_ [0-146] (Ident _id_ [10-18] "LevelOne"):
+
+            [
+                Error(
+                    Token(
+                        Close(
+                            Brace,
+                        ),
+                        Keyword(
+                            Namespace,
+                        ),
+                        Span {
+                            lo: 33,
+                            hi: 42,
+                        },
+                    ),
+                ),
+                Error(
+                    Token(
+                        Eof,
+                        Close(
+                            Brace,
+                        ),
+                        Span {
+                            lo: 163,
+                            hi: 164,
+                        },
+                    ),
+                ),
+            ]"#]],
+    );
+}
+
+#[test]
+fn namespace_with_attributes_and_docs() {
+    check_vec(
+        parse_namespaces,
+        "/// Documentation for LevelOne
+        namespace LevelOne {
+            @ExampleAttr()
+            /// Documentation for InnerItem
```

Comment:
This documentation is actually missing from the HIR below because it is getting treated as a normal comment (the parser recovery for `///` you worked on). It should appear before any attributes to correctly show up.

Reply:
Ah, that makes sense then. Maybe to help clarify intent, chance the string to something like "Documentation treated as comment" so it's clear that it's not like the one above.